system/iptables.py

```diff
@@ -292,7 +295,13 @@ def check_present(iptables_path, module, params):
 
 
 def append_rule(iptables_path, module, params):
-    cmd = push_arguments(iptables_path, '-A', params)
+    if params['order'] is not None:
+        cmd = [iptables_path]
+        cmd.extend(['-t', params['table']])
+        cmd.extend(['-I', params['chain'], params['order']])
+        cmd.extend(construct_rule(params))
+    else :
+        cmd = push_arguments(iptables_path, '-A', params)
```

Comment:
The logic here is split very strange, this function should forward stuff to `push_arguments` in both paths.